aep/general/0008/aep.md

```diff
@@ -1,5 +1,259 @@
+the guidance section is a high-level imperative, followed by an example,
+followed by a bulleted list explaining the example.
+
+### Subsection
+
+Individual subsections can be cited individually, and further elaborate
+details.
+
+## Rationale
+
+The "rationale" section is optional, and helps the reader understand the
+motivation behind specific guidance within the AEP.
+
+Deeper explanations of design justification and tradeoffs **must** be in the
+rationale instead of other sections, to ensure the rest of the document acts as
+an easily actionable reference.
+
+## History
+
+The "history" section is optional, and documents events and context around a
+significant edit to an AEP. For example, explanation of rewrite would be
+included in this section
+
+While the changelog is a dotted list of one-line summaries of changes to an AEP,
+the history section should elaborate on significant events in a descriptive
+format.
+
+The section **must not** be used to exhaustively enumerate all changes. This
+is what the changelog provides.
+
+## Further reading
+
+A bulleted list of (usually) other AEPs, in the following format:
+
+- [AEP-1](./0001.md): AEP purpose and guidelines
+
+## Changelog
+
+A bulleted list of changes in reverse chronological order, using the following
+format:
+
+- **2020-02-18**: Specified ordering.
+- **2019-07-01**: Added a subsection clarifying XYZ.
+```
+
+AEPs **should** attempt to follow this overall format if possible, but AEPs
+**may** deviate from it if necessary (in particular, if the AEP would be more
+difficult to understand, even for a reader already accustomed to reading AEPs in
+the usual format).
+
+**Note:** Except for the title, AEPs **must** only use the second heading level
+(`##`) and above. AEPs **should** only use the second and third heading levels
+(`##`, `###`).
+
+### Requirement keywords
+
+AEPs **should** use the following requirement level keywords: "MUST", "MUST
+NOT", "SHOULD", "SHOULD NOT", and "MAY", which are to be interpreted as
+described in [RFC 2119][].
+
+When using these terms in AEPs, they **must** be lower-case and **bold**. These
+terms **should not** be used in other ways.
+
+If "SHOULD" or "SHOULD NOT" are used, they **must** include valid examples of
+where other concerns may override the guidance.
+
+**Important:** If rationale is used, it exists to provide background and a more
+complete understanding, but **must not** contain guidance (and RFC-2119 terms
+**must not** be used).
+
+### Code examples
+
+API design examples in AEPs **should** be presented in both [OpenAPI][] and
+[protocol buffers][]. Examples
+**should** cover only enough syntax to explain the concept. When using RPCs in
+examples, a `library.api.http` annotation **should** be included.
+
+AEPs should use snake-case naming for parameters and properties (fields), for
+consistency across OpenAPI and protocol buffers, but other casing conventions
+may be used as long as they are applied uniformly.
```

Comment:
What about PascalCase for RPC, resource, request, and response names?

Reply:
I think we should avoid making any stand on case conventions. That is a rabbit hole.

Reply:
Just so I understand - are you saying that this AEP should be satisfied if the Get AEP says `GetBook` and the Delete AEP says `deleteBook`?

Reply:
What casing convention would allow that? "UpperLowerCamelCase"?

Sorry for the sarcasm -- the direct answer is "No". If you want we can be explicit; we can say
> You **must** establish a case convention and apply it consistently.

I'll be happy that if you like. We could even restrict this to some standard set.

Reply:
To be clear...this AEP is the AEP style guide, so I'm taking this guidance as guidance about how to write AEPs, rather than how to define APIs.  Just want to check: are we talking about the same thing?

Reply:
Indeed. So nothing else needed here, right? Please approve so we can merge.

Reply:
My original question stands:  why would the AEP style guide prescribe a casing for field names but not for e.g. method names?

Reply:
Approving and merging in advance of the barn raising; filed https://github.com/aep-dev/aep.dev/issues/129 to track this discussion.